### DISCUSSION TITLE
blog/2023-05-14-causal-inference-with-gamma-regression-or-the-problem-is-with-the-link-function-not-the-likelihood/

### DISCUSSION THREAD

**Discussion by giscus** [ASKurz/blogdown5#17] [Comments]
# blog/2023-05-14-causal-inference-with-gamma-regression-or-the-problem-is-with-the-link-function-not-the-likelihood/

Part 6 of the GLM and causal inference series.

https://solomonkurz.netlify.app/blog/2023-05-14-causal-inference-with-gamma-regression-or-the-problem-is-with-the-link-function-not-the-likelihood/

**Comment by DrJerryTAO** [comment]
"Did you know that if you take the log of a normal distribution, you end up with a lognormal distribution?" Did you mean that "Did you know that if you take the logarithm of a lognormal distribution, you end up with a normal distribution?"

**Reply by ASKurz** [reply]
Gah! Yep, you're right. Thank you for the correction. I've edited the text accordingly.

**Comment by fisher-j** [comment]
Thanks so much for writing the section on specifying a prior on the log scale equivalent to one on the identity scale--It's the only place I could find this situation described. I'm wondering though, does the method you've outlined always produce an equivalent prior?

I ask, because in my case, when modeling a strictly positive outcome using the normal distribution (identity link), I specified a prior on the intercept as $\text{Normal}(43, 65)[0, Inf]$ which resulted in a nice smooth, wide prior. But the lognormal equivalent, $\text{Lognormal}(3.323, 1.029)$ seems to be a much more specific prior, with much less density towards the tail. Thus I'm afraid that choosing $\text{Normal}(3.323, 1.029)$ with a log link is more specific than I actually intended.

Am I interpreting my the situation correctly? If I wanted a "more equivalent" prior to my first one, how would I achieve that?

![two_priors](https://github.com/ASKurz/blogdown5/assets/63136705/604737b9-de76-498b-859f-e1ec34337f7e)

**Reply by ASKurz** [reply]
Notice how the prior you're using on the left is an artificially left truncated Normal distribution. That's a very different animal from a lognormal distribution, which is naturally bounded at zero. That also means that even though the prior on the left has a mode at 43, the mean of that truncated Normal will actually be greater than 43. The issue is similar for the standard deviation. Though the SD for the Normal distribution underlying your left prior is 65, the actual SD for the truncated distribution will be different (smaller, I believe). You can find equations for the means and SDs for truncated Normal distributions here: https://en.wikipedia.org/wiki/Truncated_normal_distribution

**Reply by ASKurz** [reply]
Alternatively, if you didn't want to fool with the equations from Wikipedia, simulate like 1,000,000 draws from `rnorm(mean = 43, sd = 65)`, drop all the cases below or equal to zero, and then take the sample mean and SD for the draws that remain. Then use my formulas from above to compute the $\mu$ and $\sigma$ values for the lognormal distribution with that mean and SD.

**Reply by fisher-j** [reply]
Ha, after fooling with the equations (unsuccessfully) I did just as you recommended. This results in mean = 71 and sd = 47 which means I'd be using norm(4.1, 0.6). This definitely looks better, and I'll go with this, but it does seem that the truncated normal is still another beast :)

![two priors2](https://github.com/ASKurz/blogdown5/assets/63136705/c15aae88-fdd7-4894-962a-c028ff1f8de7)

**Reply by ASKurz** [reply]
Oh nice! Yes, agreed; the truncated normal is a different beast.

**Comment by fisher-j** [comment]
Maybe you noticed already, but your display math is not rendering in this post. Thanks again for this post, I keep coming back to it.

**Reply by ASKurz** [reply]
What browser are you using, @fisher-j? For me, the math renders on Safari just fine.

**Reply by fisher-j** [reply]
Oh, interesting. I'm on Firefox

**Reply by ASKurz** [reply]
Ah. I believe I've seen others report difficulties with math rendering properly on Firefox. I recommend using a different browser. If you don't like Safari, Google Chrome also seems to work well.

**Comment by fisher-j** [comment]
You're right! Thanks.
